jormungandr/src/rpc/eth_types/block.rs

```diff
@@ -0,0 +1,157 @@
+    fn default() -> Self {
+        Self::Hashes(Vec::new())
+    }
+}
+
+impl Serialize for BlockTransactions {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match *self {
+            BlockTransactions::Hashes(ref hashes) => hashes.serialize(serializer),
+            BlockTransactions::Full(ref ts) => ts.serialize(serializer),
+        }
+    }
+}
+
+/// Block header representation.
+#[derive(Debug, Clone, Serialize, PartialEq, Eq, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Header {
+    /// Hash of the block
+    pub hash: H256,
+    /// Mix Hash of the block
+    pub mix_hash: H256,
+    /// Nonce of the block,
+    pub nonce: U256,
+    /// Hash of the parent
+    pub parent_hash: H256,
+    /// Hash of the uncles
+    #[serde(rename = "sha3Uncles")]
+    pub uncles_hash: H256,
+    /// Alias of `author`
+    pub miner: H160,
+    /// State root hash (same as transactions_root)
+    pub state_root: H256,
+    /// Transactions root hash,
+    pub transactions_root: H256,
+    /// Transactions receipts root hash
+    pub receipts_root: H256,
+    /// Block number
+    pub number: U256,
+    /// Gas Used
+    pub gas_used: U256,
+    /// Gas Limit
+    pub gas_limit: U256,
+    /// Extra data
+    pub extra_data: Bytes,
+    /// Logs bloom
+    pub logs_bloom: Bloom,
+    /// Timestamp
+    pub timestamp: U256,
+    /// Difficulty
+    pub difficulty: Option<U256>,
+}
+
+impl Header {
+    pub fn build() -> Self {
+        Self {
+            hash: H256::zero(),
+            mix_hash: H256::zero(),
+            nonce: U256::one(),
+            parent_hash: H256::zero(),
+            uncles_hash: H256::zero(),
+            miner: H160::zero(),
+            state_root: H256::zero(),
+            transactions_root: H256::zero(),
+            receipts_root: H256::zero(),
+            number: U256::one(),
+            gas_used: U256::one(),
+            gas_limit: U256::one(),
+            extra_data: Bytes::default(),
+            logs_bloom: Bloom::zero(),
+            timestamp: U256::one(),
+            difficulty: Some(U256::one()),
+        }
+    }
+}
+
+/// Block representation
+#[derive(Debug, Serialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Block {
+    /// Header of the block
+    #[serde(flatten)]
+    pub header: Header,
+    /// Total difficulty
+    pub total_difficulty: U256,
+    /// Uncles' hashes
+    pub uncles: Vec<H256>,
+    /// Transactions
+    pub transactions: BlockTransactions,
+    /// Size in bytes
+    pub size: U256,
+    /// Base Fee for post-EIP1559 blocks.
+    pub base_fee_per_gas: Option<U256>,
+}
+
+impl Block {
+    pub fn build(full: bool) -> Self {
+        let header = Header::build();
+        let transactions = if full {
+            BlockTransactions::Hashes(vec![H256::zero()])
+        } else {
+            BlockTransactions::Full(vec![Transaction::build()])
+        };
```

Comment:
isn't the meaning of `full` the opposite of what it should be? (i.e. why the `Full` variant is returned if `full` is false?)

Reply:
Sure, mistaken